GoogleApiWrappers/src/Google.Apis.Storage.v1.ClientWrapper/DownloadObjectOptions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright 2015 Google Inc. All Rights Reserved.
+// Licensed under the Apache License Version 2.0.
+
+using Google.Apis.Download;
+
+namespace Google.Apis.Storage.v1.ClientWrapper
+{
+    /// <summary>
+    /// Options for <c>DownloadObject</c> operations.
+    /// </summary>
+    public class DownloadObjectOptions
+    {
+        // TODO: Retry policy? There's not an awful lot else to specify.
```

Comment:
How about the query parameters [here](https://cloud.google.com/storage/docs/json_api/v1/objects/get)? Of special interest I think are the `ifGenerationMatch` and `generation` parameters which will allow to download the bits only if the version is the one expected, or to download the specific version respectively. I think that specially the `IfGenerationMatch` is specially important for the Upload since that's how you implement a compare and set operation using GCS.